### PR TITLE
Convert source tests

### PR DESCRIPTION
**Pull request by emmyoop**
### Description

converts 042_sources_tests and 059_source_overrides_tests as part of the source config work.  This PR just converts the existing tests.

### Checklist

- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have added information about my change to be included in the [CHANGELOG](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry).

### CODE REVIEW

File: tests/functional/sources/test_simple_source.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+    def _create_schemas(self, project):
+        schema = self.alternative_schema(project.test_schema)
+        project.run_sql(f"drop schema if exists {schema} cascade")
+        project.run_sql(f"create schema {schema}")
+
+    def alternative_schema(self, test_schema):
+        return test_schema + "_other"
+
+    @pytest.fixture(scope="class", autouse=True)
```
Comment (gshank) [inline tests/functional/sources/test_simple_source.py:42]: Does the order of execution stay the same for this and the setUp fixture? Or doesn't it matter?

**Review by gshank — APPROVED** [review]
Looks very nice!

File: tests/functional/sources/test_simple_source.py (hunk 1 above)
Comment (emmyoop) [reply tests/functional/sources/test_simple_source.py:42]: In this case the order doesn't matter.  I think if order does matter the clearest way to enforce (and communicate) the order is to request the first fixture in the second fixture.
```
@pytest.fixture(scope="class", autouse=True)
def setUp(self, project):
    ...
@pytest.fixture(scope="class", autouse=True)
def createDummyTables(self, setUp, project):
    ...
```

File: tests/functional/source_overrides/test_simple_source_override.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        new_source_config = {
+            "sources": {
+                "localdep": {
+                    "my_other_source": {
+                        "enabled": True,
+                    }
+                }
+            }
+        }
+        update_config_file(new_source_config, project.project_root, "dbt_project.yml")
```
[the remark this answers is not in the document]
Comment (gshank) [reply tests/functional/source_overrides/test_simple_source_override.py:144]: project.py is specifically for fixtures and is loadable as a pytest plugin in the root level conftest.py. So I was thinking of util.py is for non-fixture utilities, things that we need to run during the tests.  So separating levels of concern -- project.py for test setup, and util.py for things to run.  So maybe it's fine where it is...

[the remark this answers is not in the document]
Comment (emmyoop) [reply tests/functional/source_overrides/test_simple_source_override.py:144]: By that definition I think it's fine where it is.

**Review by ChenyuLInx — APPROVED** [review]
This looks great!

**Pull request merged**